compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -5205,7 +5205,11 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign, boolean
                 matcher = IdentifierUtils.UNICODE_PATTERN.matcher(text);
             }
             if (type != SyntaxKind.TEMPLATE_STRING && type != SyntaxKind.XML_TEXT_CONTENT) {
-                text = StringEscapeUtils.unescapeJava(text);
+                try {
+                    text = StringEscapeUtils.unescapeJava(text);
+                } catch (Exception e) {
```

Comment:
Shall we capture `IllegalArgumentException` (RuntimeException). I believe there could be other IO exceptions.

Reply:
`Exception` captures `RuntimeException` and `IOException`. Do you mean to act upon them differently (i.e different error logs)? Also, [1] wraps `IOException` to `UnhandledException`. So, it's always a runtime exception that bubbles up the stack.

[1] https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/src-html/org/apache/commons/lang/StringEscapeUtils.html#line.281

Reply:
I was thinking is it okay to tell that every exception is caused by `INVALID_UNICODE`. With the context we are in I think we can assume it.